clients/web/src/dialog.js

```diff
@@ -70,24 +75,50 @@ function confirm(params) {
         yesText: 'Yes',
         yesClass: 'btn-danger',
         noText: 'Cancel',
-        escapedHtml: false
+        escapedHtml: false,
+        msgConfirmation: false,
+        additionalText: '',
+        name: ''
     }, params);
     $('#g-dialog-container').html(ConfirmDialogTemplate({
         params: params
     })).girderModal(false).one('hidden.bs.modal', function () {
         $('#g-confirm-button').off('click');
     });
 
-    var el = $('#g-dialog-container').find('.modal-body>p');
+    let el = $('#g-dialog-container').find('.modal-body>p:first-child');
     if (params.escapedHtml) {
         el.html(params.text);
     } else {
         el.text(params.text);
     }
+    if (params['msgConfirmation']) {
+        $('#g-confirm-text').css({'max-width': '50%'});
+        if (params.escapedHtml) {
+            $('.g-additional-text').html(params.additionalText);
+        } else {
+            $('.g-additional-text').text(params.additionalText);
+        }
+    }
 
     $('#g-confirm-button').off('click').click(function () {
-        $('#g-dialog-container').modal('hide');
-        params.confirmCallback();
+        if (params['msgConfirmation']) {
+            const key = `${params.yesText.toUpperCase()} ${params.name}`;
+            let msg = $('#g-confirm-text').val();
+            if (msg.toUpperCase() === key.toUpperCase()) {
+                $('#g-dialog-container').modal('hide');
+                params.confirmCallback();
+            } else if (msg.toUpperCase() === '') {
+                $('.g-msg-error').html(`Error: Tape <b>'${key}'</b>, then click on the button`);
```

Comment:
`Tape` should be `Type`.  It might be better to change this message to `Error: You need to enter <b>'${key}'</b>.`.